compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/codegen/statement.rs

```diff
@@ -99,27 +99,74 @@ impl<'tcx> GotocCtx<'tcx> {
     // TODO: this function doesn't handle unwinding which begins if the destructor panics
     // https://github.com/model-checking/rmc/issues/221
     fn codegen_drop(&mut self, location: &Place<'tcx>, target: &BasicBlock) -> Stmt {
-        // instance for drop function
         let loc_ty = self.place_ty(location);
-        let instance = Instance::resolve_drop_in_place(self.tcx, loc_ty);
-        if let Some(hk) = self.hooks.hook_applies(self.tcx, instance) {
+        let drop_instance = Instance::resolve_drop_in_place(self.tcx, loc_ty);
+        if let Some(hk) = self.hooks.hook_applies(self.tcx, drop_instance) {
```

Comment:
Do we still need the hook here? Or could we just move its logic inside the regular codegen?